File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/guild/GuildInvitedMessage.scala





















// Generated on 05/03/2013 10:34:51
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GuildInvitedMessage extends MessageDeserializer {
  val messageId = 5552

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GuildInvitedMessage.messageId

    val recruterId: Int
    val recruterName: String
    val guildInfo: BasicGuildInformations.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeInt(recruterId)
      buf.writeString(recruterName)
      guildInfo.serialize(buf)
    }

  }

  protected def create(recruterId: Int, recruterName: String, guildInfo: BasicGuildInformations.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val recruterId = buf.readInt()
    val recruterName = buf.readString()
    val guildInfo = BasicGuildInformations.deserialize(buf)

    create(recruterId, recruterName, guildInfo)
  }

}

object GuildInvitedMessage extends GuildInvitedMessage {
  case class MessageImpl(recruterId: Int, recruterName: String, guildInfo: BasicGuildInformations.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(recruterId: Int, recruterName: String, guildInfo: BasicGuildInformations.TypeImpl) = MessageImpl(recruterId, recruterName, guildInfo)

  def apply(recruterId: Int, recruterName: String, guildInfo: BasicGuildInformations.TypeImpl) = create(recruterId, recruterName, guildInfo)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GuildInvitedMessage.java


// Generated on 05/08/2013 19:37:51
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GuildInvitedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5552;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int recruterId;
    public String recruterName;
    public BasicGuildInformations guildInfo;
    
    public GuildInvitedMessage() { }
    
    public GuildInvitedMessage(int recruterId, String recruterName, BasicGuildInformations guildInfo) {
        this.recruterId = recruterId;
        this.recruterName = recruterName;
        this.guildInfo = guildInfo;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(recruterId);
        buf.writeString(recruterName);
        guildInfo.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        recruterId = buf.readInt();
        if (recruterId < 0)
            throw new RuntimeException("Forbidden value on recruterId = " + recruterId + ", it doesn't respect the following condition : recruterId < 0");
        recruterName = buf.readString();
        guildInfo = new BasicGuildInformations();
        guildInfo.deserialize(buf);
    }
    
}
